### PR TITLE
Notify on complete bug fix, Added Id to flat batch, Version Upgrades

### PR DESCRIPTION
**Pull request by TalFisApplitools**
Notify on complete bug fix
Added Id to flat batch by doing the following: -fb BATCH_NAME<>BATCH_ID
Version Upgrades:
1) eyes-images-java3 to version 3.197.0
2) pdfbox to version 2.0.23

### CODE REVIEW

**Comment by TalFisApplitools** [comment]
@yanirta One more thing.

I changed the long name of the notify on complete to `notifyCompletion` since there was a spelling error that Justin told me about.

File: src/main/java/com/yanirta/lib/TestExecutor.java [hunk 1]
```diff
@@ -54,14 +49,39 @@ public void join() {
         int curr = 1;
         while (!results_.isEmpty()) {
             try {
-                logger_.printProgress(curr++, total);
+                config_.logger.printProgress(curr++, total);
```
Comment (yanirta) [inline src/main/java/com/yanirta/lib/TestExecutor.java:52]: why?

File: src/main/java/com/yanirta/lib/TestExecutor.java [hunk 2]
```diff
@@ -54,14 +49,39 @@ public void join() {
...
                 ExecutorResult result = results_.remove().get();
-                logger_.reportResult(result);
+                config_.logger.reportResult(result);
```
Comment (yanirta) [inline src/main/java/com/yanirta/lib/TestExecutor.java:54]: why?

File: src/main/java/com/yanirta/lib/TestExecutor.java [hunk 3]
```diff
@@ -54,14 +49,39 @@ public void join() {
...
             } catch (Exception e) {
-                logger_.reportException(e);
+                config_.logger.reportException(e);
```
Comment (yanirta) [inline src/main/java/com/yanirta/lib/TestExecutor.java:56]: why?

File: src/test/java/Tests/TestFullCommands.java [hunk 4]
```diff
@@ -77,28 +77,46 @@ public void testBatchNotifications() {
...
+
+    @Test
+    public void testBatchNotificationsWithAll() {
+        ImageTester.main("-f  TestData/ -nc".split(" "));
+    }
+
     @Test
     public void testImageScaling1() {
-        ImageTester.main("-f TestData/a -ms 1000".split(" "));
+        ImageTester.main("-f TestData/a/ -ms 1000".split(" "));
```
Comment (yanirta) [inline src/test/java/Tests/TestFullCommands.java:105]: why?

File: src/test/java/Tests/TestFullCommands.java [hunk 5]
```diff
@@ -77,28 +77,46 @@ public void testBatchNotifications() {
...
     @Test
     public void testImageScaling3() {
-        ImageTester.main("-f TestData/a -ms 1000x1000".split(" "));
+        ImageTester.main("-f TestData/a/ -ms 1000x1000".split(" "));
     }
 
     @Test
     public void testImageScaling4() {
-        ImageTester.main("-f TestData/a -ms 1000x".split(" "));
+        ImageTester.main("-f TestData/a/ -ms 1000x".split(" "));
```
Comment (yanirta) [inline src/test/java/Tests/TestFullCommands.java:120]: why??

**Review by yanirta — APPROVED** [review]
Some flavor rejects, nothing with high impact.

**Pull request merged**